packages/picasso/src/PageSidebar/use-sidebar-state/use-sidebar-state.tsx

```diff
@@ -0,0 +1,27 @@
+import { useSidebar } from '@toptal/picasso-provider'
+import { useState } from 'react'
+
+export interface Props {
+  defaultCollapsed?: boolean
+}
+const useSidebarState = ({ defaultCollapsed }: Props) => {
+  const useSidebarProps = useSidebar()
+
+  const [expandedItemKey, setExpandedItemKey] = useState<number | null>(null)
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(
+    Boolean(defaultCollapsed)
+  )
+  const [isHovered, setIsHovered] = useState<boolean>(false)
```

Comment:
Nit: You don't need to type `useState` generics here, Typescript can infer the type from the expression

```suggestion
  const [isCollapsed, setIsCollapsed] = useState(
    Boolean(defaultCollapsed)
  )
  const [isHovered, setIsHovered] = useState(false)
```

Another nit is that I prefer using `!!` instead of calling `Boolean`, but is very minor